snipssonos/services/entities_injection_service.py

```diff
@@ -12,21 +12,25 @@ def __init__(self, hermes_host):
         self.mqtt_client = MqttClient(hermes_host)
         self.mqtt_client.run()
 
-    def publish_entities(self, entity_slot_name, data):
-        payload = self.build_payload(entity_slot_name, data)
+        self.entities_payload = dict()
+
+    def publish_entities(self):
+        payload = self.build_payload()
 
         injection_topic = self.MQTT_TOPIC_INJECT
         self.mqtt_client.publish(injection_topic, payload)
+        self.entities_payload = dict()
 
-    def build_payload(self, entity_slot_name, data):
-        entities_payload = dict()
+    def build_entities_payload(self, entity_slot_name, data):
         parsed_data = self.parse_data(entity_slot_name, data)
-        entities_payload[entity_slot_name] = parsed_data
-        logging.info("Injecting data: %s", parsed_data)
+        self.entities_payload[entity_slot_name] = parsed_data
```

Comment:
Nazi time ! 
For me, this introduces a side effect. The use case should not have to know about how the injection service works. 
It should not have to know that you have to loop over the entity types and build a payload for all the types. 

The `InjectionService` should have a method that takes a dict of entities types, and loop through them to get build the payload.

Reply:
That sounds right in theory but (maybe) the problem is that you are dependant on the music customization service to get the right data to fill the payload at each iteration of the loop. Does CLEAN allow services to be calling other services?

Reply:
Your service is the implementation of a interface. If your interface specify a dependency on another service, then you have then you have the right to import one service into another service :) !

Reply:
Nope, it is not an implementation of an interface (should it?). So to clarify: if I want to comply with CLEAN design, services should be implementations of interfaces and interface will define how I can use my service in conjunction with other services, right?

Reply:
Let's talk about it IRL